### PR TITLE
LPS-181482 Changing Object Definition ERC results in Duplicate Table error toast

### PR DESCRIPTION
**Pull request by diogomelo09**
[LPS-181482](https://issues.liferay.com/browse/LPS-181482)

### CODE REVIEW

**Comment by diogomelo09** [comment]
ci:test:relevant

**Comment by diogomelo09** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Hey @cgmonte , 
Could you review this PR please? :smile:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx [hunk 1]
```diff
@@ -205,6 +205,13 @@ export default function EditObjectDetails({
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [objectDefinitionId]);
 
+	const onExternalReferenceCodeChange = (externalReferenceCode: string) => {
+		setValues({
+			...values,
```
Comment (cgmonte) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx:210]: You don´t have to spread values when using the `setValues` returned by `useObjectDetailsForm`. **Please, remove the spread of `values`**. `useObjectDetailsForm` is a custom hook, programmed to easy the state update calls, different from a stock hook like `setState`, where most of the time you will have to use spreads and other magic to ensure a new object has the updated value as well as the old ones.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx [hunk 2]
```diff
@@ -222,6 +229,9 @@ export default function EditObjectDetails({
 						values.name
 					)}
 					objectDefinitionId={objectDefinitionId}
+					onExternalReferenceCodeChange={
```
Comment (cgmonte) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx:232]: If you look at this file, you will see that `setValues` is being passed to most of its child. Follow the pattern, pass `setValues` to `ObjectManagementToolbar` and do something like this: 
```
<ManagementToolbar
  ...
  onExternalReferenceCodeChange={() => setValues({externalReferenceCode})} 
  ...
/>
```

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hey @diogomelo09, Thanks for sending this!! I made some suggestions in comments, please let me know if they make sense to you. Reach at Slack if you need any help.

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by cgmonte** [comment]
ci:forward

**Comment by cgmonte** [comment]
ci:reevaluate:1397401_4544

**Comment by cgmonte** [comment]
ci:forward:force

**Pull request closed without merging**